js/src/viewer.js

```diff
@@ -23,8 +23,10 @@
         'bookmarkPanelVisible': false
       },
       manifests:             [] 
-    }, $.DEFAULT_SETTINGS, options);
+    }, options);
 
+    this.id = this.state.getStateProperty('id');
+    this.data = this.state.getStateProperty('data');
```

Comment:
Here are a few examples where we are getting state information from the saveController